HCF-15 modify User (add tel and cart), add UserDto.

lyubovkirpa opened the pull request:
HCF-15 modify User (add tel and cart), add UserDto.  Add changelog create table bed, room, cart.

lyubovkirpa committed:
HCF-12 modify User (add tel and cart), add UserDto. Add changelog create table bed, room

lyubovkirpa committed:
HCF-15 modify UserDto.

AIT-Reutow commented on src/main/java/ait/cohort49/hostel_casa_flamingo/model/dto/UserDto.java:
```diff
@@ -0,0 +1,106 @@
+    private String lastName;
+    private String email;
+    private String tel;
+    private Set<Role> roles = new HashSet<>();
```
Желательно так же использовать DTO и для ролей

lyubovkirpa committed:
HCF-15 add UserMappingService.

lyubovkirpa committed:
HCF-20 add RoleDto.

AIT-Reutow committed:
Add PostgreSQL Docker setup with persistent volume

AIT-Reutow committed:
Add GitHub Actions workflow for Java CI with Maven

This workflow automates building Java projects using Maven on push and pull request events. It runs on Ubuntu, sets up JDK 17, and caches Maven dependencies for faster builds. Environment variables are configured for build customization.

AIT-Reutow committed:
Update GitHub Actions to use latest dependencies

Upgraded actions/checkout to v4 and actions/setup-java to v4. Updated JDK version from 17 to 21 for compatibility with newer Java features. Cleaned up unused environment variables in the Maven build step.

AIT-Reutow committed:
Update build workflow to target specific branches

Modified the GitHub Actions workflow to trigger on `push` and `pull_request` events only for the `main` and `dev` branches. This prevents unnecessary builds for other branches and optimizes CI pipelines.

AIT-Reutow committed:
Add separate test step to GitHub Actions workflow

This change introduces a distinct step for running tests after building the project in the CI pipeline. It ensures clarity and separation of concerns within the workflow, improving maintainability and debugging efficiency.

AIT-Reutow committed:
Add test configurations and dependencies

Introduced an `application-test.yaml` file for test environment setup and updated the `pom.xml` with required dependencies for testing, including H2 database and JUnit. Also activated the test profile in `HostelCasaFlamingoApplicationTests` by adding the `@ActiveProfiles("test")` annotation.

AIT-Reutow committed:
Add test configurations and dependencies

Introduced an `application-test.yaml` file for test environment setup and updated the `pom.xml` with required dependencies for testing, including H2 database and JUnit. Also activated the test profile in `HostelCasaFlamingoApplicationTests` by adding the `@ActiveProfiles("test")` annotation.

AIT-Reutow committed:
Rename build file and refine Maven workflow steps

Renamed the GitHub Actions workflow file for better clarity. Updated Maven workflow to separate build and test steps, skipping tests during the build phase for efficiency. This improves overall readability and simplifies task execution.

AIT-Reutow committed:
Merge branch 'master' into 15-Entity-User

# Conflicts:
#	src/main/resources/db/changelog/changelog-master.xml
#	src/main/resources/db/changelog/v0.2.2/changelog.xml

AIT-Reutow committed:
Remove cart-related tables, relationships, and references.

AIT-Reutow committed:
Merge branch 'master' of https://github.com/lyubovkirpa/hostel_Casa_Flamingo into 15-Entity-User
